guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/wherefore-art-thou/index.md

```diff
@@ -145,6 +145,31 @@ Check equivalence of `Object` in `collection` with `Object` passed as second par
 *   <a>Array.prototype.reduce()</a>
 *   <a href='https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty' target='_blank' rel='nofollow'>Object.hasOwnProperty()</a>
 
+## ![:rotating_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/rotating_light.png?v=3 ":rotating_light:") Advanced Code Solution:
+
+    function whatIsInAName(collection, source) {
+        return collection.filter(a =>
+    		Object.keys(source).every(key => a[key] === source[key])
```

Comment:
```suggestion
    Object.keys(source).every(key => a[key] === source[key])
```

Reply:
Please use variable names which describe the content instead of something generic like `a` which conveys no meaning.

Reply:
Will do